MainModule/Client/Core/Functions.lua

```diff
@@ -873,30 +873,36 @@ return function(Vargs, GetEnv)
 				Variables.LightingSettings[prop] = value
 			end
 		end;
-
-		ChatMessage = function(msg,color,font,size)
-			local tab = {}
-
-			tab.Text = msg
-
-			if color then
-				tab.Color = color
-			end
-
-			if font then
-				tab.Font = font
-			end
-
-			if size then
-				tab.Size = size
-			end
-
-			service.StarterGui:SetCore("ChatMakeSystemMessage",tab)
-
-			if Functions.SendToChat then
+	
+		ChatMessage = function(msg:string,color,font,size)
+		local tab = {}
+		local FormattedColor = Color3.new(1, 1, 1)
+		tab.Text = msg
+
+		if color then
+			tab.Color = color
+			FormattedColor  = Color3.new(color):ToHex() -- // This line is here for TextChatService support.
+		end
+
+		if font then
+			tab.Font = font
+		end
+
+		if size then
+			tab.Size = size
+		end
+		if game:GetService('TextChatService').ChatVersion == Enum.ChatVersion.TextChatService then
+			game:GetService('TextChatService').TextChannels.RBXGeneral:DisplaySystemMessage(string.format("<font color='%s'>%s</font>",FormattedColor,msg))
```

Comment:
Don't forget the hashtag "#" character before the HEX color code! otherwise it won't work as expected here...

Reply:
Ok

Reply:
yeha that i forgot when removing a function